src/CoreWCF.Primitives/src/CoreWCF/Runtime/ReadOnlySequenceExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace CoreWCF.Runtime;
+
+public static class ReadOnlySequenceExtensions
+{
+    public static unsafe string ParseAsUTF8String(this ReadOnlySequence<byte> slice, int stringLength)
+    {
+        const int MaxStackAllocSize = 128;
+        var decoder = Encoding.UTF8.GetDecoder();
+        var preProcessedBytes = 0;
+        var processedCharacters = 0;
+
+        char[] rented = null;
+        Span<char> characterSpan = stringLength > MaxStackAllocSize
+            ? rented = ArrayPool<char>.Shared.Rent(stringLength)
+            : stackalloc char[MaxStackAllocSize];
+        try
+        {
+            foreach (var memory in slice)
+            {
+                preProcessedBytes += memory.Length;
+                bool isLast = preProcessedBytes == slice.Length;
+                var emptyCharSlice =
+                    characterSpan.Slice(processedCharacters, stringLength - processedCharacters);
+
```

Comment:
Both stackalloc and the rented allocation could be (probably are) longer than stringLength. If you pre-slice characerSpan with Slice(0, stringLength), then here you can just use characterSpan.Slice(preprocessedCharacters) and not provide the length.

Reply:
good catch thanks i fixed it